R/class-Megastudy.R

```diff
@@ -0,0 +1,97 @@
+check_study_vocabulary <- function(object) {
+  errors <- character()
+
+  if (is.na(object@studyIdColumnName)) {
+    errors <- c(errors, "StudyIdColName is required but not provided.")
+  }
+
+  if (is.na(object@variableSpec@variableId)) {
+    errors <- c(errors, "VariableSpec is required but not provided.") 
+  }
+
+  return(if (length(errors) == 0) TRUE else errors)  
+}
+
+#' @include class-VariableMetadata.R
+#' @export
+StudySpecificVocabulary <- setClass("StudySpecificVocabulary", representation(
```

Comment:
so the megastudy data comes in as some large matrix thing with a column for study id? Then columns are variables? And study id is like a variable basically?

Reply:
im anticipating letting the ancestor ids be read into r, though we usually wouldnt. study is an entity node in the megastudy.. if we wanted to make it a bit more agnostic, so it could handle future possible megastudies w the same needs, we could call it root node rather than study.